aws-datastore/src/main/java/com/amplifyframework/datastore/storage/LocalStorageAdapter.java

```diff
@@ -132,6 +132,25 @@ void query(
             @NonNull Consumer<DataStoreException> onError
     );
 
+    /**
+     * Deletes all items of a given type from storage that meet the
+     * specific conditions. A {@link Consumer} will be invoked when the
+     * save operation is completed, to notify the caller of success or failure.
+     * @param <T> The type of item being deleted
+     * @param itemClass Item to delete
+     * @param initiator An identification of the actor who initiated this deletion
+     * @param predicate Predicate condition for conditional delete
+     * @param onSuccess A callback that will be invoked when deletion succeeds
+     * @param onError A callback that will be invoked when deletion fails with an error
+     */
+    <T extends Model> void delete(
```

Comment:
I think we should try to _reduce_ the number of APIs we have on the local storage adapter.

Would it be possible to consolidate our various public `delete(...)` behaviors, so that they call one single primitive method on the storage adapter?

We want to provide various APIs on the public DataStore contract, to make things easier for customers. But to help keep our storage layer lean and mean, it would be better to centralize its entry points.